nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,438 @@
 """
-Visualizing 3D stat maps in a Papaya viewer
+Visualizing 3D stat maps in a Brainsprite viewer
 """
+import warnings
 import os
-import tempfile
-import base64
 import json
+from io import BytesIO
+from base64 import encodebytes
+import numbers
 
 import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.cm as mpl_cm
+from matplotlib.image import imsave
+from matplotlib import colors
 
-from .. import image, datasets
+from nibabel.affines import apply_affine
+
+from ..image import resample_img, resample_to_img, new_img_like
+from .js_plotting_utils import get_html_template, HTMLDocument
+from ..plotting import cm
+from ..plotting.find_cuts import find_xyz_cut_coords
+from ..plotting.img_plotting import _load_anat, _get_colorbar_and_data_ranges
 from .._utils import check_niimg_3d
 from .._utils.extmath import fast_abs_percentile
-from .._utils.param_validation import check_threshold
-from .js_plotting_utils import get_html_template, HTMLDocument, colorscale
-
-
-def _encode_nii(img):
-    # downcast to float32 to save memory
-    img = image.load_img(img, dtype='auto')
-    fd, temp = tempfile.mkstemp(suffix='.nii')
-    os.close(fd)
-    try:
-        img.to_filename(temp)
-        with open(temp, 'rb') as f:
-            binary = f.read()
-            encoded = base64.b64encode(binary).decode('utf-8')
-            return encoded
-    finally:
-        os.remove(temp)
-
-
-def _decode_nii(encoded):
-    fd, temp = tempfile.mkstemp(suffix='.nii')
-    os.close(fd)
-    try:
-        with open(temp, 'wb') as f:
-            f.write(base64.b64decode(encoded.encode('utf-8')))
-        img = image.load_img(temp)
-        # The asarray/copy is to get rid of memmapping
-        loaded = image.new_img_like(img,
-                                    np.asarray(img.get_data()).copy())
-        del img
-        return loaded
-    finally:
-        os.remove(temp)
+from .._utils.niimg import _safe_get_data
+from ..datasets import load_mni152_template
+
+
+def _resample_to_self(img, interpolation):
+    """ For internal use.
+        Resample an image to have a diagonal, isotropic affine.
+    """
+
```

Comment:
Maybe you can start the function by using nilearn.image.reorder_img: it can avoid more costly resampling if the image is only a diagonal affine permuted. I would do first a "reorder_img(..., resample=None), and then your logic, if the resulting affine is not isotropic diagonal.

Reply:
Done. Will need testing.

Reply:
Actually, what I have implemented is already in reorder_img!! I read the doc of that function too fast, and thought it only reordered/flipped axes. But I ended up looking at the code to check that it indeed flipped axes, and I realized that with the resample option it does exactly what I wanted. Great news, less code to test and read!